src/modules/rlm_sql/rlm_sql.c

```diff
@@ -779,6 +779,86 @@ static int sql_get_grouplist(rlm_sql_t const *inst, rlm_sql_handle_t **handle, R
 	return num_groups;
 }
 
+static rlm_rcode_t sql_get_grouplist_resume(REQUEST *request, void *instance, void *thread, void *ctx);
+
+/*
+ * Get group list
+ * Yield if SQL driver supports async queries.
+ */
+static rlm_rcode_t sql_get_grouplist_async(REQUEST *request, void *instance, void *thread, void *ctx)
+{
+	rlm_sql_thread_group_t *group_ctx = ctx;
```

Comment:
The sql_thread_group structure is there. Why do you say it's missing?
It holds the context for the group related queries.

I used the _async suffix because in some cases, I kept the original implementation to make the transition work easier. The goal is to remove it at some point.